src/core/Header/HeaderLogo.tsx

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import '@itwin/itwinui-css/css/header.css';
+
+import cx from 'classnames';
+import React from 'react';
+
+import { useTheme } from '../utils/hooks/useTheme';
+import { CommonProps } from '../utils/props';
+
+export type HeaderLogoProps = {
+  /**
+   * Logo shown before the main content.
+   */
+  logo: JSX.Element;
+  /**
+   * Click event handler.
+   * Will update the Logo to have mouse and keyboard interaction if provided.
+   */
+  onClick?: () => void;
+  /**
+   * Children is put on the right of the logo.
+   */
+  children?: React.ReactNode;
+} & CommonProps;
+
+/**
+ * Header Title section
+ * @example
+ * <HeaderLogo logo={<SvgImodelHollow />}>iTwin Application</HeaderLogo>
+ * <HeaderLogo logo={<SvgImodelHollow />} />
+ * <HeaderLogo logo={<img src='image.png' />} />
+ * <HeaderLogo logo={<img src='data:image/png;base64,...' />} >Downloaded Image</HeaderLogo>
+ */
+export const HeaderLogo = (props: HeaderLogoProps) => {
+  const { className, children, logo, onClick, ...rest } = props;
+  const keyDownHandler = (e: React.KeyboardEvent<HTMLDivElement>) => {
+    if (onClick && (e.key === 'Enter' || e.key === ' ')) {
```

Comment:
Also check for `Spacebar` because some browsers return that instead of `' '`.

Reply:
That's one of the reason I did not want to handle that at all... but documentation does not mention "spacebar" as a valid key: https://www.w3.org/TR/uievents-key/#keys-whitespace

Reply:
Spacebar is used in older browsers 
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values#:~:text=older%20browsers%20may%20return%20%22spacebar%22%20instead%20of%20%22%20%22

Reply:
Added